### PR TITLE
Update portability-analyzer.md

### LINKED ISSUE #12985: Need clarification of how to interpret the Portability analyzer report
The doc doesn't have instruction of how interpret the Portability analyzer report . Some of the existing link or concepts can have some fix up as well. 

---
#### Document Details

⚠ *Do not edit this section. It is required for docs.microsoft.com ➟ GitHub issue linking.*

* ID: cdd8086b-9a95-38d5-160f-c12c653f954e
* Version Independent ID: 0f47d826-e3ef-8549-9d37-2e8afae29f46
* Content: [The .NET Portability Analyzer - .NET](https://docs.microsoft.com/en-us/dotnet/standard/analyzers/portability-analyzer)
* Content Source: [docs/standard/analyzers/portability-analyzer.md](https://github.com/dotnet/docs/blob/master/docs/standard/analyzers/portability-analyzer.md)
* Product: **dotnet**
* Technology: **dotnet-standard**
* GitHub Login: @dotnet-bot
* Microsoft Alias: **dotnetcontent**

### PR DESCRIPTION
**Pull request by Lxiamail**
## Summary

Updated .NET portability analyzer doc to clarify some concepts, add more explanation (especially for how to interpret the report), and fixed some links.

Fixes #12985

### CODE REVIEW

File: docs/standard/analyzers/portability-analyzer.md
Comment (mairaw) [inline docs/standard/analyzers/portability-analyzer.md]: This isn't really separate from the Report Summary, is it? It seems like a children section of the Report Summary.

File: docs/standard/analyzers/portability-analyzer.md
Comment (mairaw) [inline docs/standard/analyzers/portability-analyzer.md]: you need to improve the formatting here. You can see how it's rendering right now:
https://review.docs.microsoft.com/en-us/dotnet/standard/analyzers/portability-analyzer?branch=pr-en-us-12986#details

File: docs/standard/analyzers/portability-analyzer.md
Comment (mairaw) [inline docs/standard/analyzers/portability-analyzer.md]:
```suggestion
In Visual Studio, under the **Messages** tab in the **Error List**, it displays missing APIs with caller file and line number. You can jump to problem areas directly from the **Messages** tab.
```

File: docs/standard/analyzers/portability-analyzer.md
Comment (mairaw) [inline docs/standard/analyzers/portability-analyzer.md]: Where's that column? Do we need to run this against a better sample?

There are some typos here (It normally are, your analyzed assembly reference to, portability analyze run, looks for). Do you want to review this and I can propose some changes after I see your intention.

**Review by mairaw — DISMISSED** [review]
Thanks @Lxiamail for updating this article. I've left some initial feedback for you.

File: docs/standard/analyzers/portability-analyzer.md
Comment (Lxiamail) [reply docs/standard/analyzers/portability-analyzer.md]: @Mariaw thanks for the quick response!

File: docs/standard/analyzers/portability-analyzer.md
Comment (Lxiamail) [reply docs/standard/analyzers/portability-analyzer.md]: I think it still belongs to "How to use the .NET Portability Analyzer". Since "How to use the .NET Portability Analyzer" is H2, this should be H3.

File: docs/standard/analyzers/portability-analyzer.md
Comment (Lxiamail) [reply docs/standard/analyzers/portability-analyzer.md]: It is separated section from Report Summary. The separation is not obvious in VS report. In the excel report, it is on a separated worksheet.

File: docs/standard/analyzers/portability-analyzer.md
Comment (marklio) [inline docs/standard/analyzers/portability-analyzer.md]: I think this should say .NET Core?

```suggestion
* .NET Core + [Platform Extensions](../../core/porting/windows-compat-pack.md): Includes the .NET Core APIs in addition to the Windows Compatibility Pack, which provides many of .NET Framework available technologies, so it's much easier to build .NET Core applications and .NET Core libraries
```

File: docs/standard/analyzers/portability-analyzer.md
Comment (marklio) [inline docs/standard/analyzers/portability-analyzer.md]: Should this mention apps at all?

```suggestion
* .NET Standard + [Platform Extensions](../../core/porting/windows-compat-pack.md): Includes the .NET Standard APIs in addition to the Windows Compatibility Pack, which provides many of .NET Framework available technologies, so it's much easier to build .NET Standard libraries.
```

File: docs/standard/analyzers/portability-analyzer.md
Comment (marklio) [inline docs/standard/analyzers/portability-analyzer.md]:
```suggestion
To begin using the .NET Portability Analyzer in Visual Studio, you first need to download and install the extension from the [Visual Studio Marketplace](https://marketplace.visualstudio.com/items?itemName=ConnieYau.NETPortabilityAnalyzer). It works on Visual Studio 2017 and later versions. You can configure it in Visual Studio via **Analyze** > **Portability Analyzer Settings** and select your Target Platforms, which is the .NET platforms/versions that you want to evaluate the portability gaps comparing with the platform/version that your current assembly is built with.
```

Comment (marklio) [reply docs/standard/analyzers/portability-analyzer.md]: I don't know why the above comment doesn't highlight the diff. "assemblies build with" should either be "assembly's built with", "assembly is built with", or "assemblies are built with".

File: docs/standard/analyzers/portability-analyzer.md
Comment (marklio) [inline docs/standard/analyzers/portability-analyzer.md]: Is there not a direct download link for the tool somewhere?

File: docs/standard/analyzers/portability-analyzer.md
Comment (marklio) [inline docs/standard/analyzers/portability-analyzer.md]:
```suggestion
To analyze your entire project in Visual Studio, right-click on your project in **Solution Explorer** and select **Analyze Assembly Portability**. Otherwise, go to the **Analyze** menu and select **Analyze Assembly Portability**. From there, select your project’s executable or DLL.
```

**Review by marklio — APPROVED** [review]
Made a few suggestions. This seems strictly better.

File: docs/standard/analyzers/portability-analyzer.md
Comment (Lxiamail) [reply docs/standard/analyzers/portability-analyzer.md]: No, the tool is not released anywhere other than the github repo.

Comment (marklio) [reply docs/standard/analyzers/portability-analyzer.md]: Might I suggest we point directly to https://github.com/microsoft/dotnet-apiport/releases?

Comment (marklio) [reply docs/standard/analyzers/portability-analyzer.md]: Apparently, @terrajobst owns http://aka.ms/apiport. I suggest we use that, and perhaps request co-ownership of that link.

Comment (marklio) [reply docs/standard/analyzers/portability-analyzer.md]: hmmm. it doesn't point to releases. I still think we should use it, and maybe update the text to indicate you should download it from the releases tab. Or make aka.ms/apiportdownload. I'll leave it up to you. I've made too many comments :)

File: docs/standard/analyzers/portability-analyzer.md
Comment (Lxiamail) [reply docs/standard/analyzers/portability-analyzer.md]: @mairaw  Yes, we need update the sample image. Where should I upload the new sample images?

File: docs/standard/analyzers/portability-analyzer.md
Comment (Lxiamail) [reply docs/standard/analyzers/portability-analyzer.md]: fixed

File: docs/standard/analyzers/portability-analyzer.md
Comment (Lxiamail) [reply docs/standard/analyzers/portability-analyzer.md]: Actually, the VS report image will need to be updated because the VS extension latest release had removed the VS report UI and used link to excel report file anyway. Once the report image is updated. This issue is not an concern any more.

File: docs/standard/analyzers/portability-analyzer.md
Comment (Lxiamail) [reply docs/standard/analyzers/portability-analyzer.md]: Great idea! Updated to use https://aka.ms/apiportdownload for console app download.

**Comment by Lxiamail** [comment]
Updated portability report images to replace previously out of date image. Also changed "New target" section to "Common target" and added more guidance about when to use which target. @mairaw and @marklio can you help to do the final review before merge?

File: docs/standard/analyzers/portability-analyzer.md
Comment (twsouthwick) [inline docs/standard/analyzers/portability-analyzer.md]: The download I believe only comes with .NET Framework at the moment.

File: docs/standard/analyzers/portability-analyzer.md
Comment (twsouthwick) [inline docs/standard/analyzers/portability-analyzer.md]: Can you add something to the effect that you only want to submit user assemblies?

File: docs/standard/analyzers/portability-analyzer.md
Comment (twsouthwick) [inline docs/standard/analyzers/portability-analyzer.md]: This header is a bug - there's an issue for it that we discussed and @Lxiamail was going to verify is already fixed (not sure what the status is). Needs to be ingested into the service

File: docs/standard/analyzers/portability-analyzer.md
Comment (twsouthwick) [inline docs/standard/analyzers/portability-analyzer.md]: The third sentense reads weird. Should may be: These are assemblies that your analyzed assembly references but could not be found by the analyzer.

Also, a comment to the fact that there is nothing wrong with having missing assemblies; they just weren't analyzed. We've discussed renaming it to "Unresolved Assemblies" instead for that reason.

File: docs/standard/analyzers/portability-analyzer.md
Comment (Lxiamail) [reply docs/standard/analyzers/portability-analyzer.md]: Good catch, fixed by removing the statement.

File: docs/standard/analyzers/portability-analyzer.md
Comment (Lxiamail) [reply docs/standard/analyzers/portability-analyzer.md]: Added some guidance about this

File: docs/standard/analyzers/portability-analyzer.md
Comment (Lxiamail) [reply docs/standard/analyzers/portability-analyzer.md]: Fixed the header

File: docs/standard/analyzers/portability-analyzer.md
Comment (Lxiamail) [reply docs/standard/analyzers/portability-analyzer.md]: Updated to address the feedback.

**Comment by Lxiamail** [comment]
@mairaw Thanks for all the feedbacks. Have the updates addressed all your concerns?

File: docs/standard/analyzers/portability-analyzer.md
Comment (Thraka) [inline docs/standard/analyzers/portability-analyzer.md]:
```suggestion
* [ASP.NET Core](/aspnet/core): A modern web-framework built on .NET Core. If your goal is to port your web app to .NET Core to support multiple platforms, this is the recommended target.
```

File: docs/standard/analyzers/portability-analyzer.md
Comment (Thraka) [inline docs/standard/analyzers/portability-analyzer.md]:
```suggestion
* .NET Core + [Platform Extensions](../../core/porting/windows-compat-pack.md): Includes the .NET Core APIs in addition to the Windows Compatibility Pack, which provides many of the .NET Framework available technologies. This is a recommended target for porting your app from .NET Framework to .NET Core on Windows.
```

File: docs/standard/analyzers/portability-analyzer.md
Comment (Thraka) [inline docs/standard/analyzers/portability-analyzer.md]:
```suggestion
* .NET Standard + [Platform Extensions](../../core/porting/windows-compat-pack.md): Includes the .NET Standard APIs in addition to the Windows Compatibility Pack, which provides many of the .NET Framework available technologies. This is a recommended target for porting your library from .NET Framework to .NET Core on Windows.
```

File: docs/standard/analyzers/portability-analyzer.md
Comment (Thraka) [inline docs/standard/analyzers/portability-analyzer.md]: You should have something under the ## and not just another header like ###. I honestly don't think you need `### Pick Target Platforms`. I think the intro paragraph below gives enough information without needed an extra header. You can still keep the other headers below like `### Analyze portability`
```suggestion

```

File: docs/standard/analyzers/portability-analyzer.md
Comment (Thraka) [inline docs/standard/analyzers/portability-analyzer.md]: I would introduce this as an alternative to visual studio... `If you do not want to use Visual Studio, ...`

There is also a double space between `can  use`

Also, use the \` inline code markdown on `listtargets`, `-t`, and `--target`

File: docs/standard/analyzers/portability-analyzer.md
Comment (Thraka) [inline docs/standard/analyzers/portability-analyzer.md]: Again, introduce this as an alternative, much like the text you changed indicated.

File: docs/standard/analyzers/portability-analyzer.md
Comment (Thraka) [inline docs/standard/analyzers/portability-analyzer.md]: I'm confused about these `\...\`

File: docs/standard/analyzers/portability-analyzer.md
Comment (Thraka) [inline docs/standard/analyzers/portability-analyzer.md]: Again, what is this \`...\` commands? These seem wrong??

File: docs/standard/analyzers/portability-analyzer.md
Comment (Thraka) [inline docs/standard/analyzers/portability-analyzer.md]:
```suggestion
After running the analysis in Visual Studio, you'll see your .NET Portability report file link pops up. If you used the [ApiPort console app](https://aka.ms/apiportdownload), your .NET Portability report is saved as a file in the format you specified. The default is in an Excel file (*.xlsx*) in your current directory.
```

File: docs/standard/analyzers/portability-analyzer.md
Comment (Thraka) [inline docs/standard/analyzers/portability-analyzer.md]: These screenshots introduce a large problem. They should be in all lowercase. You must do the following to fix this, simply renaming them in Windows will not work with git.

1. Copy the pictures to a temp directory.
2. Delete the pictures here in the repo.
3. Commit the delete changes.
4. Rename the pictures to all lowercase characters
5. Commit the "new" files.

This creates two commits in your git history, one to delete the file and another to recreate it in a new name format.

Then you need to edit this article and make sure the name of the files match the lowercase file name.

**Review by Thraka — CHANGES_REQUESTED** [review]

**Comment by Thraka** [comment]
@Lxiamail Hi! Maira is on vacation. I left some feedback about issues I noticed.

File: docs/standard/analyzers/portability-analyzer.md
Comment (Lxiamail) [reply docs/standard/analyzers/portability-analyzer.md]: Fixed

File: docs/standard/analyzers/portability-analyzer.md
Comment (Lxiamail) [reply docs/standard/analyzers/portability-analyzer.md]: Since these two parallel approaches, instead of preferring VS tension. I tried to avoid using "don't want" kind of term. Does it make sense?  But I'm not English native speaker, I would like to hear your suggestion. :)

File: docs/standard/analyzers/portability-analyzer.md
Comment (Lxiamail) [reply docs/standard/analyzers/portability-analyzer.md]: I guess it means absent of whatever relative path from current directory. I'm not the original author. Any suggestion to make it clearer?

File: docs/standard/analyzers/portability-analyzer.md
Comment (Thraka) [reply docs/standard/analyzers/portability-analyzer.md]: I understand, good idea! How about `You can also use the ApiPort console application ...`

File: docs/standard/analyzers/portability-analyzer.md
Comment (Thraka) [reply docs/standard/analyzers/portability-analyzer.md]: I would just remove it. I don't see what benefit it gives. We always just specify the name of the command directly. I think people using the command line are familiar enough with pathing to understand that if it's not in PATH or in the current directory, they would have to specify it.

File: docs/standard/analyzers/portability-analyzer.md
Comment (Lxiamail) [reply docs/standard/analyzers/portability-analyzer.md]: fixed

File: docs/standard/analyzers/portability-analyzer.md
Comment (Lxiamail) [reply docs/standard/analyzers/portability-analyzer.md]: fixed

File: docs/standard/analyzers/portability-analyzer.md
Comment (Lxiamail) [reply docs/standard/analyzers/portability-analyzer.md]: fixed

File: docs/standard/analyzers/portability-analyzer.md
Comment (Lxiamail) [reply docs/standard/analyzers/portability-analyzer.md]: fixed

File: docs/standard/analyzers/portability-analyzer.md
Comment (Thraka) [inline docs/standard/analyzers/portability-analyzer.md]:
```suggestion

# The .NET Portability Analyzer
```

**Comment by Lxiamail** [comment]
@Thraka Do you have any other concern?

**Comment by Thraka** [comment]
@Lxiamail the screenshots needed a gray border so the white areas don't blend in. I added that to each screenshot and committed that for you 😄 

I'm going to merge it. Awesome job!

**Review by Thraka — APPROVED** [review]

**Pull request merged**
